components/notifier/src/notifier_utilities/data.py

```diff
@@ -0,0 +1,114 @@
+"""Database interaction functions for the notifier."""
+
+import os
+from typing import Optional
+
+import pymongo
+from shared.initialization.database import create_indexes
+from shared.initialization.datamodel import import_datamodel
+from shared.initialization.report import initialize_reports_overview
+from shared.initialization.secrets import initialize_secrets
+from shared.model.metric import Metric
+from shared.model.report import Report
+from shared_data_model import DATA_MODEL
+
+
+def client(url: Optional[str] = "mongodb://root:root@localhost:27017") -> pymongo.MongoClient:
+    """return a pymongo client
+
+    Args:
+        url (str, optional): url to mongodb. Defaults to "mongodb://root:root@localhost:27017".
+
+    Returns:
+        pymongo.MongoClient: pymongo client
+    """
+    database_url = os.environ.get("DATABASE_URL", url)
+    return pymongo.MongoClient(database_url)
+
+
+def database(url: Optional[str] = "mongodb://root:root@localhost:27017") -> pymongo.database.Database:
+    """Return a pymongo database.
+
+    Args:
+        url (str, optional): url to mongodb. Defaults to "mongodb://root:root@localhost:27017".
+
+    Returns:
+        pymongo.database.Database: quality-time database
+    """
+    db_client = client(url)
+    create_indexes(db_client['quality_time_db'])
+    import_datamodel(db_client['quality_time_db'])
+    initialize_secrets(db_client['quality_time_db'])
+    initialize_reports_overview(db_client['quality_time_db'])
```

Comment:
I understand this is here because the internal server used to do this, but do we really need to have this initialisation in three components: the (external) server, the notifier, and later also the collector?